src/main/src/main/java/org/geoserver/security/urlchecks/GeoServerURLChecker.java

```diff
@@ -50,12 +52,34 @@ public boolean confirm(String url) {
 
             // if enabled but no checks configured, deny access
             List<AbstractURLCheck> enabledUrlList = getEnabledChecks();
-            if (enabledUrlList.isEmpty()) return false;
-
-            for (AbstractURLCheck u : enabledUrlList) {
-                if (u.confirm(url)) {
-                    LOGGER.log(Level.FINE, () -> "URL " + url + " was matched by " + u);
-                    return true;
+            if (!enabledUrlList.isEmpty()) {
+                // Check 1: Check using test URL provided
+                for (AbstractURLCheck u : enabledUrlList) {
+                    if (u.confirm(url)) {
+                        LOGGER.log(Level.FINE, () -> "URL " + url + " was matched by " + u);
+                        return true;
+                    }
+                }
+                // Check 2: Check using normalized test URL if different
+                try {
+                    String normalized = new URI(url).normalize().toString();
+                    if (!normalized.equals(url)) {
+                        for (AbstractURLCheck u : enabledUrlList) {
+                            if (u.confirm(normalized)) {
+                                LOGGER.log(
+                                        Level.FINE,
+                                        () ->
+                                                "URL "
+                                                        + url
+                                                        + " was normalized to "
+                                                        + normalized
+                                                        + " and matched by "
+                                                        + u);
```

Comment:
Deep nesting is visible at odds with our coding conventions, please avoid. Thinking out loud, the loop could be a separate method accepting a Function, first you run it with the identity function, then with one normalizing the URL.

That said... this normalization is not where it should be... GeoTools code using URL checks in other applications would end up suffering the same problem. Better to do it in URLCheckers instead?